liblumen_alloc/src/erts/process/trace/backtrace.rs

```diff
@@ -0,0 +1,338 @@
+    }
+
+    #[inline]
+    pub fn print(&self, kind: Term, reason: Term) -> std::io::Result<()> {
+        format::print(self, kind, reason)
+    }
+
+    #[inline]
+    pub fn format(&self, f: &mut fmt::Formatter, kind: Term, reason: Term) -> std::io::Result<()> {
+        format::format(self, f, kind, reason)
+    }
+
+    /// Sets the top frame of the stacktrace to a specific module/function/arity,
+    /// using the provided argument list in place of arity. This is a special case
+    /// added to support `undef` or `badarg` errors, which may display the arguments
+    /// used to call the function which raised the error.
+    ///
+    /// NOTE: Support for this is optional, we are not required to display the arguments,
+    /// but may do so if available. At a minimum, we do need to support the ability to
+    /// add a frame to the trace for calls which we know will fail (e.g. apply/3 with
+    /// a function that doesn't exist). This still feels like a gross hack though.
+    #[inline]
+    pub fn set_top_frame(&self, mfa: &ModuleFunctionArity, arguments: &[Term]) {
+        assert!(self.top.is_none(), "top of trace was already set");
+
+        // Get heap to allocate the frame on
+        let sizeof_args: usize = arguments
+            .iter()
+            .map(|t| t.size_in_words() * mem::size_of::<Term>())
+            .sum();
+        let extra = utils::BASE_FRAME_SIZE + sizeof_args;
+        let heap_ptr = self.get_or_create_fragment(extra).unwrap_or(None);
+        if let Some(mut heap) = heap_ptr {
+            let heap_mut = unsafe { heap.as_mut() };
+            if let Ok(frame) = utils::format_mfa(heap_mut, mfa, Some(arguments), None, None) {
+                unsafe {
+                    self.top.set(Some(frame));
+                }
+            }
+        }
+    }
+
+    #[inline]
+    pub fn push_frame(&mut self, frame: &Frame) {
+        unsafe {
+            self.frames.as_mut().push(TraceFrame::from(frame));
+        }
+    }
+
+    pub fn as_term(&self) -> AllocResult<Term> {
+        if let Some(term) = self.term.as_ref() {
+            Ok(*term)
+        } else {
+            self.construct()
+        }
+    }
+
+    #[inline(always)]
+    pub fn into_raw(trace: Arc<Trace>) -> *mut Trace {
+        Arc::into_raw(trace) as *mut Trace
+    }
+
+    #[inline(always)]
+    pub unsafe fn from_raw(trace: *mut Trace) -> Arc<Trace> {
+        Arc::from_raw(trace)
+    }
+}
+
+pub struct SymbolIter<'a> {
+    frames: &'a [TraceFrame],
+    top: Option<Term>,
+    pos: Option<usize>,
+}
+impl<'a> SymbolIter<'a> {
+    fn new(frames: &'a [TraceFrame], top: Option<Term>) -> Self {
+        Self {
+            frames,
+            top,
+            pos: None,
+        }
+    }
+}
+
+impl Iterator for SymbolIter<'_> {
+    type Item = Symbolication;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        use std::convert::TryInto;
+
+        println!("{:?}", &self.pos);
```

Comment:
These two `println!` look like they are for debug purposes.  Should they be removed?